US-CBP/cbp-theme
# Extends webpack build tools to the 'ds-guide', with additional improvements

#### What's this PR do?

Basically this enhances the overall build and testing processes being used in both dirs (ds-css, ds-ux-guidelines) by: 

1.  creating and placing all the distributed css assets within the 'ds-css' directory
2. places all necessary CSS files within a folder at 'ds-ux-guidelines/css' that contains all needed CSS assets for local development.
3. creates and manages the linting processes for both directories
4. creates and manages the accessibility testing process within the 'ds-ux-guidelines' dir
5. provides a development files watching tool

Detailed breakdown:

Both directories have their own separate build processes:

1. for 'ds-css' --> npm run build
(* please see the scripts w/in the package.json)
2. for 'ds-ux-guidelines' --> npm run build-guide
(* please see the scripts w/in the package.json)

#### Where should the reviewer start?

Can begin by looking at the updates made to the npm scripts being used for everything within both directories package.json files.

#### How should this be manually tested?

First, just to make sure all tools are installed correctly for both 'ds-css' & 'ds-ux-guidelines' directories after you pull in updates, you can:
` rm -rf node_modules ` & ` rm package-lock.json ` & ' npm i '

Next, can just run:

` npm run build ` for the (ds-css) directory
` npm run build-guide ` for the (ds-ux-guidelines) directory

While it's running, can watch the changes populated to both the 'ds-css' and 'ds-ux-guidelines' directories.

#### Any background context you want to provide?

We wanted to improve upon our existing build to provide a more optimized local development process while making sure to correctly maintain our distributed css assets.

#### What are the relevant tickets?
n/a
#### Screenshots (if appropriate)
n/a
#### Questions:
- Is there a blog post?
n/a
- Does the knowledge base need an update?
n/a
